src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala

```diff
@@ -199,10 +199,51 @@ class ModeChoiceMultinomialLogit(
     attributesOfIndividual: Option[AttributesOfIndividual],
     destinationActivity: Option[Activity]
   ): Double = {
-    val waitingTime = embodiedBeamTrip.totalTravelTimeInSecs - embodiedBeamTrip.legs.map(_.beamLeg.duration).sum
+    getGeneralizedTimeOfTripInHours(embodiedBeamTrip, attributesOfIndividual, destinationActivity)
+  }
+
+  private def getGeneralizedTimeOfTripInHours(
+    embodiedBeamTrip: EmbodiedBeamTrip,
+    attributesOfIndividual: Option[AttributesOfIndividual],
+    destinationActivity: Option[Activity],
+    adjustSpecialBikeLines: Boolean = false
+  ): Double = {
+    val adjustedTripDuration = if (adjustSpecialBikeLines && embodiedBeamTrip.tripClassifier == BIKE) {
+      calculateBeamTripTimeInSecsWithSpecialBikeLanesAdjustment(embodiedBeamTrip)
+    } else {
+      embodiedBeamTrip.legs.map(_.beamLeg.duration).sum
+    }
+    val waitingTime: Int = embodiedBeamTrip.totalTravelTimeInSecs - adjustedTripDuration
+    embodiedBeamTrip.legs.map { x: EmbodiedBeamLeg =>
+      val factor = if (adjustSpecialBikeLines) {
+        bikeLanesAdjustment.bikeLaneScaleFactor(x.beamLeg.mode, adjustSpecialBikeLines)
+      } else {
+        1D
+      }
+      getGeneralizedTimeOfLeg(x, attributesOfIndividual, destinationActivity) * factor
+    }.sum + getGeneralizedTime(waitingTime, None, None)
+  }
+
+  private def calculateBeamTripTimeInSecsWithSpecialBikeLanesAdjustment(
+    embodiedBeamTrip: EmbodiedBeamTrip
+  ): Int = {
     embodiedBeamTrip.legs
-      .map(x => getGeneralizedTimeOfLeg(x, attributesOfIndividual, destinationActivity))
-      .sum + getGeneralizedTime(waitingTime, None, None)
+      .map { embodiedBeamLeg: EmbodiedBeamLeg =>
+        pathScaledForWaiting(embodiedBeamLeg.beamLeg.travelPath)
+      }
+      .sum
+      .toInt
+  }
+
+  private def pathScaledForWaiting(path: BeamPath): Double = {
```

Comment:
nitpick: having tests for this will be helpful in the future (can be extracted to the object and just unit tested this part)